include/openssl/ssl.h

```diff
@@ -547,7 +547,7 @@ typedef int (*SSL_verify_cb)(int preverify_ok, X509_STORE_CTX *x509_ctx);
 # define SSL_CONF_TYPE_NONE              0x4
 
 /* Length of a TLSv1.3 cookie */
-# define SSL_COOKIE_LENGTH                       255
+# define SSL_COOKIE_LENGTH                       65535
```

Comment:
> Cookies can now be up to 2^16-1 bytes.

You are confusing 2 different things:

1) The length of the overall cookie
2) The length of that bit of the cookie which an application can add via the callback set in `SSL_CTX_set_cookie_generate_cb()`.

It is the second item which this define describes. In truth perhaps it could be given a better name to avoid this confusion. Also SSL_CTX_set_cookie_generate_cb() is a function used for DTLS that is being reused for TLSv1.3 cookies, but it is missing documentation. We should probably add some...or maybe even use a different callback altogether for this as it is subtly different.

> It's unclear to me if DTLS1_COOKIE_LENGTH should also be updated, or whether there are other less obvious places that need changing as well for this to work in practice.

Nothing had changed wrt to DTLS cookie length so that value needs to stay the same. We couldn't change it anyway because it is in a public header file, so changing the value would break binary compatibility.

Reply:
I see; thanks for the detailed explanation! The coincidental matching of the old cookie size threw me off, but reviewing `tls_parse_ctos_cookie` and `tls_construct_stoc_cookie` a second time was enlightening.

It still seems strange that the limit is so much lower than what the RFC permits. Would I be correct to conjecture that app cookies are expected in practice to always be a MAC of a client's source address, and that 256 bytes is hence expected to be sufficient for the forseeable future?

I'd be happy to convert this PR into one that adds a man3 document for the cookie callbacks if that would be helpful, although I'm not sure I have a firm enough grasp on these APIs to do a great job of it just yet.

Reply:
The thinking was that the cookie size returned by `SSL_CTX_set_cookie_generate_cb()` should always be the same regardless of the context (TLS1.3 or DTLS). Although note that even here there is an issue. `DTLS1_COOKIE_LENGTH` is mistakenly set to 256 bytes, but the actual maximum is 255 bytes (the current value of SSL_COOKIE_LENGTH). Unfortunately we can't fix it because that would be a binary incompatible change.

I do begin to wonder whether a new callback would be better, rather than trying to reuse the old DTLS one.

Anyway, please feel free to have a go at writing a PR if you want to. I will happily review it.